packages/hardhat-core/test/internal/hardhat-network/provider/node.ts

```diff
@@ -817,7 +817,31 @@ describe("HardhatNode", () => {
     });
 
     describe("shouldn't break snapshotting", async function () {
-      it("when doing mineBlocks() before a snapshot", async function () {});
+      it("when doing mineBlocks() before a snapshot", async function () {
+        await node.mineBlocks(new BN(10));
+
+        const latestBlockNumberBeforeSnapshot = node.getLatestBlockNumber();
+
+        const snapshotId = await node.takeSnapshot();
+        await node.sendTransaction(
+          createTestTransaction({
+            from: DEFAULT_ACCOUNTS_ADDRESSES[1],
+            to: Address.fromString(
+              "0x1111111111111111111111111111111111111111"
+            ),
+            gasLimit: 21000,
+          })
+        );
+        await node.mineBlocks(new BN(1));
```

Comment:
Is this line necessary? Shouldn't it be auto-mining? Even if it is necessary, it feels weird to have it here, since you could just call `mineBlock()` instead, and since `mineBlocks(1)` literally just calls `mineBlock()` for you, and since seeing `mineBlocks()` within the test kinda suggests that it's essential to the test (it's not...right?), since `mineBlocks()` is the thing under test.

Reply:
I think this `node` instance is created with `automine: false` at the top-level `describe`.